innovation_sweet_spots/pipeline/fetch_daps1_data/daps1_utils.py

```diff
@@ -0,0 +1,88 @@
+# Generic scripts to get DAPS tables
+import logging
+import os
+from configparser import ConfigParser
+from typing import Any, Dict, Iterator
+
+import pandas as pd
+from sqlalchemy import create_engine
+from sqlalchemy.engine.url import URL
+
+# from pandas._typing import FilePathOrBuffer  # Not available in pandas < 1
+
+MYSQL_CONFIG = os.environ["MYSQL_CONFIG"]
+
+
+def get_engine(config_path, database="production", **engine_kwargs):
+    """Get a SQL alchemy engine from config"""
+    cp = ConfigParser()
+    cp.read(config_path)
+    cp = cp["client"]
+    url = URL(
+        drivername="mysql+pymysql",
+        database=database,
+        username=cp["user"],
+        host=cp["host"],
+        password=cp["password"],
+    )
+    return create_engine(url, **engine_kwargs)
+
+
+def fetch_daps_table(table_name: str, fields: str = "all") -> pd.DataFrame:
+    """Fetch DAPS tables if we don't have them already
+    Args:
+        table_name: name
+        path: path for the table
+        fields: fields to fetch. If a list, fetches those
+    Returns:
+        table
+    """
+    logging.info(f"Fetching {table_name}")
+    engine = get_engine(MYSQL_CONFIG)
+    con = engine.connect().execution_options(stream_results=True)
+
+    if fields == "all":
+        chunks = pd.read_sql_table(table_name, con, chunksize=1000)
+    else:
+        chunks = pd.read_sql_table(table_name, con, columns=fields, chunksize=1000)
```

Comment:
Avoid option like `fields` which are string values - generally it reads a bit like an old fashioned `C` `case` statement 😄 

In this case the `pandas` default for columns (`None`) will return all fields, so just use that:

```python
def fetch_daps_table(table_name: str, columns: list = None) -> pd.DataFrame:
    """Fetch DAPS tables if we don't have them already
    Args:
        table_name: name
        path: path for the table
        columns: columns to fetch. If a list, fetches those
    Returns:
        table
    """
    logging.info(f"Fetching {table_name}")
    engine = get_engine(MYSQL_CONFIG)
    con = engine.connect().execution_options(stream_results=True)
    return pd.read_sql_table(table_name, con, columns=columns, chunksize=1000)
```